ckolderup/postmarks
# Empty title fixes; formatting AP content updates

Resolution for #143

# Testing
I have the site running with these changes at: 

https://smooth-fish-dash.glitch.me/
https://glitch.com/edit/#!/smooth-fish-dash

Testing scenarios below. If no title is entered and not found via opengraph, title should be empty.  Easiest way to do this is use an invalid URL so OpenGraph doesn't work.

* Attempt to add bookmark without url 
* Add from website @ /add
* Add from Bookmarklet widget @ /admin
* Add from Multi Add import @ /admin/bookmarks
* Edit from website @ /bookmark/{id}/edit

If no url was added on the Add or removed from Edit and then submitted:

* Should not be allowed and browser specific required message should appear

If the title isn't added, the results should be:

* ActivityPub outbox message includes url as the title
* Bookmark display @ root should display url as title
* Bookmark display @ /bookmark/{id} should display url as title
* Bookmark display @ /bookmark/{id}/edit should NOT display url as title, it should show blank
* Bookmark entry in XML feed @ /index.xml should display url as title
* Atom Feed should be valid, even when title is empty ([validator](https://validator.w3.org/feed/)

In addition:
* ActivityPub message should not have an indention on the description as viewed in Mastodon iOS, Ice Cubes, Ivory, Mona and Elk.
* ActivityPub link (title/url) in the content should have the privacy attributes and when clicked, open in a new window

Pull request incoming.
